pkg/resource/parameter_group/hooks.go

```diff
@@ -277,3 +271,23 @@ func sdkTagsFromResourceTags(
 	}
 	return tags
 }
+
+func resourceTagsFromSDKTags(
+	sdkTags []*svcsdk.Tag,
+) []*svcapitypes.Tag {
+	tags := make([]*svcapitypes.Tag, len(sdkTags))
+	for i := range sdkTags {
+		tags[i] = &svcapitypes.Tag{
+			Key:   sdkTags[i].Key,
+			Value: sdkTags[i].Value,
+		}
+	}
+	return tags
+}
+
+func equalStrings(a, b *string) bool {
+	if a == nil {
+		return b == nil || *b == ""
+	}
+	return (*a == "" && b == nil) || *a == *b
+}
```

Comment:
This might not panic in the tags case, it could with other fields. Correct implementation here: https://github.com/aws-controllers-k8s/community/issues/1654

Reply:
I will fix it for all resource hooks in the new PR.

Reply:
You should be able to just copy the code from that issue into all of your `equalStrings` functions. It would be better if we address it now than leave it for another PR, where we might forget.

Reply:
Fixed it for all hook files.